core/src/console/console.cc

```diff
@@ -1144,22 +1143,40 @@ static void TerminateConsole(int sig)
 static int CheckResources()
 {
   bool OK = true;
-  DirectorResource* director;
-
   ResLocker _{my_config};
 
-  g_numdir = 0;
-  foreach_res (director, R_DIRECTOR) { g_numdir++; }
-
-  if (g_numdir == 0) {
+  std::size_t director_count = 0;
+  for (BareosResource* console = nullptr;
+       (console = my_config->GetNextRes(R_DIRECTOR, console));) {
+    ++director_count;
+  }
+  if (director_count == 0) {
     const std::string& configfile_name = my_config->get_base_config_path();
     Emsg1(M_FATAL, 0,
           T_("No Director resource defined in %s\n"
-             "Without that I don't how to speak to the Director :-(\n"),
+             "Without that I don't know how to speak to the Director :-(\n"),
           configfile_name.c_str());
     OK = false;
   }
 
+  std::size_t console_count = 0;
+  for (BareosResource* console = nullptr;
+       (console = my_config->GetNextRes(R_CONSOLE, console));) {
+    ++console_count;
+  }
+
+  if (console_count == 0) {
```

Comment:
as we only care about having 0 consoles, we could simply do
```suggestion
  if (my_config->GetNextRes(R_CONSOLE, nullptr) == nullptr) {
```

Reply:
actually the same is true for the director `if()` as well, so you may as well change it there too.

Reply:
as we actually need access to the director resource anyways, maybe it makes sense to just define these right after the `ResLocker`:

```C++
  ResLocker _{my_config};
  auto* director = dynamic_cast<DirectorResource*>(my_config->GetNextRes(R_DIRECTOR, nullptr));
  auto* console = dynamic_cast<ConsoleResource*>(my_config->GetNextRes(R_CONSOLE, nullptr));
```

Then we can simply do `if (!director)` and `if (!console)`